pkg/kubelet/kubeletconfig/util/codec/codec.go

```diff
@@ -81,7 +83,7 @@ func DecodeKubeletConfiguration(kubeletCodecs *serializer.CodecFactory, data []b
 	// the UniversalDecoder runs defaulting and returns the internal type by default
 	obj, gvk, err := kubeletCodecs.UniversalDecoder().Decode(data, nil, nil)
 	if err != nil {
-		return nil, fmt.Errorf("failed to decode, error: %v", err)
+		return nil, errors.Wrap(err, "failed to decode")
```

Comment:
Are there any guidelines on when to use errors.Wrap vs. fmt.Errorf with the new %w verb?

Reply:
i don't think we have guidelines for %w yet.

errors.Wrap can be useful for:
- having a consistent wrapping for errors "some message: error"
- stack traces.

Reply:
Didn't notice that `Wrap` was from a third-party lib during my first review pass, rather than the std lib (same package name...). We should probably be using the format string approach, since that seems to be the built-in way to do it. https://github.com/golang/go/wiki/ErrorValueFAQ

Reply:
the third-party library announced that it's getting deprecated after the recent changes in go, so we should be moving away from it.
however i'm not seeing a way to consistently wrap the format string:

> fmt.Errorf("more info: %w", err)
vs 
> errors.Wrap(err, "more info")

Reply:
we aren't on go 1.13